tests/unit/test_source.py

```diff
@@ -1,22 +1,92 @@
 import pytest
 import vorta.views
+from PyQt6 import QtCore
+from PyQt6.QtWidgets import QMessageBox
 
 
-def test_add_folder(qapp, qtbot, mocker, monkeypatch, choose_file_dialog):
-    monkeypatch.setattr(vorta.views.source_tab, "choose_file_dialog", choose_file_dialog)
-    main = qapp.main_window
-    main.tabWidget.setCurrentIndex(1)
-    tab = main.sourceTab
+@pytest.fixture()
+def source_env(qapp, qtbot, monkeypatch, choose_file_dialog):
+    def setup():
+        monkeypatch.setattr(vorta.views.source_tab, "choose_file_dialog", choose_file_dialog)
+        main = qapp.main_window
+        main.tabWidget.setCurrentIndex(1)
+        tab = main.sourceTab
+        qtbot.waitUntil(lambda: tab.sourceFilesWidget.rowCount() == 1, timeout=2000)
+        return main, tab
 
+    main, tab = setup()
+    yield main, tab
+
+    # Wait for directory sizing to finish
+    qtbot.waitUntil(lambda: len(qapp.main_window.sourceTab.updateThreads) == 0, timeout=2000)
```

Comment:
After some digging, I discovered this wait needs to be done after each source test that involves `updateThreads`. I added this to the `source_env` fixture to happen at teardown.